### PR TITLE
Added an Example for BLEU.

### PR DESCRIPTION
**Pull request by Neeshamraghav012**
I would like to inform you that I have made some updates to the content. Firstly, I have added an example for BLEU which was not present before. This new example serves as an illustration of the concept and should be helpful for those who are new to this topic.

Furthermore, I have also fixed the URLs by wrapping them in markdown. This should make it easier for readers to access the resources mentioned in the content.

### CODE REVIEW

**Comment by Neeshamraghav012** [comment]
@abheesht17 I have deleted the previous pull request because I messed up with it. Please review this one.

File: keras_nlp/metrics/bleu.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # you may not use this file except in compliance with the License.
 # You may obtain a copy of the License at
 #
-#     https://www.apache.org/licenses/LICENSE-2.0
+#       https://www.apache.org/licenses/LICENSE-2.0
```
Comment (abheesht17) [inline keras_nlp/metrics/bleu.py:7]: Please remove extra spaces.

File: keras_nlp/metrics/bleu.py [hunk 2]
```diff
@@ -92,7 +92,86 @@ class Bleu(keras.metrics.Metric):
...
-        **kwargs: Other keyword arguments.
+
+    Examples:
+
+    1. Various Input Types.
+    1.1. Python string.
+    >>> bleu = keras_nlp.metrics.Bleu(max_order=4)
+    >>> ref_sentence = "the quick brown fox jumps over the lazy dog"
+    >>> pred_sentence = "the quick brown fox jumps over the box"
+    >>> score = bleu([ref_sentence], [pred_sentence])
```
Comment (abheesht17) [inline keras_nlp/metrics/bleu.py:103]: Since we mention that this example is with Python string inputs, this should ideally be 

```
bleu([ref_sentence], pred_sentence)
```

File: keras_nlp/metrics/bleu.py [hunk 3]
```diff
@@ -92,7 +92,86 @@ class Bleu(keras.metrics.Metric):
...
+
+    1.2. rank 1 inputs.
+    a. Python list.
+    >>> bleu = keras_nlp.metrics.Bleu(max_order=4)
+    >>> ref_sentence = [
+        "the quick brown fox jumps over the lazy dog",
+        "the quick brown fox jumps over the lazy frog"
+    ]
+    >>> pred_sentence = ["the quick brown fox jumps over the box"]
+    >>> score = bleu(ref_sentence, pred_sentence)
```
Comment (abheesht17) [inline keras_nlp/metrics/bleu.py:114]: Instead of assigning the score to a variable, let's just have 

```
>>> bleu(ref_sentence, pred_sentence)
<tf.Tensor(0.7420885, shape=(), dtype=float32)>
```

Please do the same in other places as well.

File: keras_nlp/metrics/bleu.py [hunk 4]
```diff
@@ -92,7 +92,86 @@ class Bleu(keras.metrics.Metric):
...
+        [
+            "the quick brown fox jumps over the lazy dog",
+            "the quick brown fox jumps over the lazy frog"
+        ]
+    ])
+    >>> pred_sentence = tf.ragged.constant([
+        ["the quick brown fox jumps over the box"]
+    ])
+    >>> score = bleu(ref_sentence, pred_sentence)
+    <tf.Tensor(0.7420885, shape=(), dtype=float32)>
```
Comment (abheesht17) [inline keras_nlp/metrics/bleu.py:139]: Since we mention that we are using ragged tensors here, it would be great if we could actually have a tensor with variable dimension along an axis. Also, I think your example will throw an error; `pred_sentence` should be a dense tensor here, not a ragged tensor.

So, something like:

```
>>> bleu = keras_nlp.metrics.Bleu(max_order=4)
>>> ref_sentence = tf.ragged.constant([
...         ["the quick brown fox jumps over the lazy dog"],
...         ["the quick brown fox jumps over the lazy dog", "the quick brown fox jumps over the lazy frog"]
...     ])
>>> pred_sentence = tf.constant([
...         ["the quick brown fox jumps over the box"],
...         ["the quick brown fox jumps over the box"]
...     ])
>>> bleu(ref_sentence, pred_sentence)
<tf.Tensor: shape=(), dtype=float32, numpy=0.7420885>
```

File: keras_nlp/metrics/bleu.py (hunk 3 above)
Comment (abheesht17) [reply keras_nlp/metrics/bleu.py:114]: Although the above example works, the ideal input formula is:

```
rank(ref_sentence) = rank(pred_sentence) + 1
```

I think we should probably stick to this formula in our examples. Otherwise, it becomes confusing for the reader. Could you please make this change everywhere?

File: keras_nlp/metrics/bleu.py [hunk 5]
```diff
@@ -92,7 +92,86 @@ class Bleu(keras.metrics.Metric):
...
+
+    1. Various Input Types.
+    1.1. Python string.
+    >>> bleu = keras_nlp.metrics.Bleu(max_order=4)
+    >>> ref_sentence = "the quick brown fox jumps over the lazy dog"
+    >>> pred_sentence = "the quick brown fox jumps over the box"
+    >>> score = bleu([ref_sentence], [pred_sentence])
+    <tf.Tensor(0.7420885, shape=(), dtype=float32)>
+
+    1.2. rank 1 inputs.
```
Comment (abheesht17) [inline keras_nlp/metrics/bleu.py:106]: We should change these to 

```
rank 1 prediction inputs, rank 2 reference inputs
```

or something like that?

**Review by abheesht17 — CHANGES_REQUESTED** [review]
Left a few comments. Most of them our minor, but there is one major comment regarding rank of inputs!

**Comment by abheesht17** [comment]
@mattdangerw - for context, this is the previous PR: https://github.com/keras-team/keras-nlp/pull/799

File: keras_nlp/metrics/bleu.py (hunk 5 above)
Comment (Neeshamraghav012) [reply keras_nlp/metrics/bleu.py:106]: I checked the Rouge L metric documentation, where this same convention is used. Is it okay to make these changes?

**Comment by sachinprasadhs** [comment]
@Neeshamraghav012 , Is this PR still relevant, if yes, could you please update the status of this PR. Thanks
